controllers/topics/index.js

```diff
@@ -189,17 +218,30 @@ const followTopicV2 = async (req, res) => {
   }
 };
 
-const _afterPutTopic = async (topic, token, userId, name) => {
+const _afterPutTopic = async (isUnfollow, token, userId, name) => {
   // follow / unfollow main feed topic
-  if (topic) {
+  if (isUnfollow) {
     await unfollowMainFeedTopic(token, userId, name);
   } else {
     await followMainFeedTopic(token, userId, name);
   }
 
-  const message = topic ? 'Success delete topic user v2' : 'Success add topic user v2';
+  let data;
+  if (isUnfollow) {
+    data = {
+      message: 'Success delete topic user v2',
+      type: 'unfollow',
+      userId
+    };
+  } else {
+    data = {
+      message: 'Success add topic user v2',
+      type: 'follow',
+      userId
+    };
+  }
 
-  return message;
+  return data;
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

This function `_afterPutTopic` has been refactored to return an object instead of a string message. This change improves the maintainability of the code by making it easier to handle the response data. However, the `userId` field in the returned object might expose sensitive user information. Consider removing this field if it's not necessary.

```diff
-      userId
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->